net/init-metrics.sh

```diff
@@ -83,11 +83,16 @@ else
   query "CREATE DATABASE \"$netBasename\""
   query "ALTER RETENTION POLICY autogen ON \"$netBasename\" DURATION 7d"
   query "GRANT READ ON \"$netBasename\" TO \"ro\""
-  query "GRANT WRITE ON \"$netBasename\" TO \"scratch_writer\""
+  query "GRANT WRITE ON \"$netBasename\" TO \"${username}\""
 
-  SOLANA_METRICS_CONFIG="host=$host,db=$netBasename,u=scratch_writer,p=topsecret"
+  SOLANA_METRICS_CONFIG="host=$host,db=$netBasename,u=${username},p=${password}"
+  set +x
+  echo "Consider adding \`SOLANA_METRICS_CONFIG\` to your shell environment for future use:"
+  echo "export SOLANA_METRICS_CONFIG=\"$SOLANA_METRICS_CONFIG\""
 fi
 
+set -x
+sed -i '/^export\s\+SOLANA_METRICS_CONFIG\s*=\s*"/d' "${configFile}"
```

Comment:
why do we need this now?

Reply:
yes, this make sure that calling init-metrics several times will not produce multiple lines in the config. AFAIK only the last line will actually be used, but having several there looks like something went wrong.

Reply:
okay, so it does not really change anything.

but this is a regex, and are you sure this will not erroneously delete something else in the future?

Reply:
I've tested the regex with several examples and it worked exactly as needed (though I am aware how horrible regex can be). If it was production code in agave I'd not use regex at all. However, the config file here is not going to lose people money or do any serious harm. Also in this case I see no real alternative to sed be honest.

Reply:
so my suggestion is why to bother with it to begin with when it does not fix anything, and being a regex, it can only waste people time in the future should it break things.

Reply:
Without regex script leaves multiple copies of the variable in the config which looks like a bug if one opens said config. Even though it works correctly.

Reply:
okay, but that is still not a bug.
bash has well-defined behavior if a variable is exported several times and it has many legit use-cases that this script might also utilize in the future.

I still think it is best to remove this because it does not solve any problems, it adds confusion here, and potentially can erroneously delete something which it shouldn't in the future.

Reply:
Done, sed line removed as suggested.

Reply:
> okay, but that is still not a bug. bash has well-defined behavior if a variable is exported several times and it has many legit use-cases that this script might also utilize in the future.

Isn't it just confusing if your config contains multiple, potentially duplicate lines that specify the metrics environment variable?

I can not think if a legitimate use case this can have in the future.
Can provide an example?
Something that would justify not cleaning things correctly right now.
 
> I still think it is best to remove this because it does not solve any problems, it adds confusion here, and potentially can erroneously delete something which it shouldn't in the future.

The regex is pretty accurate.
It only targets lines that start with `export` followed by `SOLANA_METRICS_CONFIG =`.
Considering the bash syntax, it seems to be as good as it gets.